files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharp.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Text;
+
+namespace FastestWayToReadATextFileInCsharp
+{
+    public class WaysToReadATextFileInCsharp
+    {
+        private readonly string _sampleFilePath;
+
+        public WaysToReadATextFileInCsharp(string sampleFilePath)
+        {
+            _sampleFilePath = sampleFilePath;
+        }
+
+        public string UseFileReadAllLines()
+        {
+            var stringArray = File.ReadAllLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseFileReadAllText() => File.ReadAllText(_sampleFilePath);
+        
+        public string UseFileReadLines()
+        {
+            var stringArray = File.ReadLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseStreamReaderReadLine()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var stringBuilder = new StringBuilder();
+            string fileLine;
+
+            while ((fileLine = streamReader.ReadLine()) != null)
+            {
+                stringBuilder.AppendLine(fileLine);
+            }
+
+            return stringBuilder.ToString().TrimEnd();
+        }
+
+        public string UseStreamReaderReadToEnd()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+
+            return streamReader.ReadToEnd();
+        }
+
+        public string UseStreamReaderReadBlock()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var buffer = new char[1024];
+            int numberRead;
+            var stringBuilder = new StringBuilder();
+
+            while ((numberRead = streamReader.ReadBlock(buffer, 0, buffer.Length)) > 0)
+            {
+                stringBuilder.Append(buffer, 0, numberRead);
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        public string UseBufferedStreamObject()
+        {
+            var stringBuilder = new StringBuilder();
+
+            using (var fileStream = new FileStream(_sampleFilePath, FileMode.Open))
+            using (var bufferedStream = new BufferedStream(fileStream))
+            using (var streamReader = new StreamReader(bufferedStream))
+            {
+                string fileLine;
+
+                while ((fileLine = streamReader.ReadLine()) != null)
+                {
+                    stringBuilder.AppendLine(fileLine);
+                }
```

Comment:
Could we just use StreamReader.ReadToEnd here? I would think it would be more performant.

Reply:
@jshergal the article is about benchmarking different ways, so it's important to have this one and also, ReadToEnd in another method.

Reply:
@cjaynjoku I think you can skip this one.

Reply:
@karimkod - If we leave this with the `AppendLine` then we have the potential issue of text fidelity between the line endings in the file being read and the line endings in the result. This may be fine under some use cases, but it for sure needs to be pointed out in the article that there is a potential loss of fidelity.

Reply:
@jshergal - the `StreamReader.ReadLine` [doc ](https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.readline?view=net-7.0#remarks) states that : 

> A line is defined as a sequence of characters followed by a line feed ("\n"), a carriage return ("\r"), or a carriage return immediately followed by a line feed ("\r\n"). **The string that is returned does not contain the terminating carriage return or line feed. The returned value is null if the end of the input stream is reached.**

The line endings in the file won't be read. Would the risk still be present?